core/src/core_tests/local_activities.rs

```diff
@@ -884,6 +884,96 @@ async fn test_schedule_to_start_timeout_not_based_on_original_time(
     worker.run_until_done().await.unwrap();
 }
 
+#[rstest::rstest]
+#[tokio::test]
+async fn start_to_close_timeout_allows_retries(#[values(true, false)] la_completes: bool) {
```

Comment:
Why not make this an integration test instead of crafting a history?

Reply:
A few reasons:
* Speed, it's nice to keep integration test suite small, because that means it's fast. If it doesn't need to be an IT, I can keep that. Running this test by itself is (slightly) faster too w/o the server start/stop.
* Easier to use w/ a debugger
* What needs to happen here is clear, so I can pretty easily test just that